CrashReport 3.1.3: ClassCastException getActivityComponent

**Describe the bug**
while trying to perform injection the application returned is not our application class, maybe null?

**Expected behavior**
Injection should happen with no problem

**Logs**
```
Caused by: java.lang.ClassCastException: 

 
  at org.kiwix.kiwixmobile.ActivityExtensionsKt.getKiwixComponent (ActivityExtensionsKt.java:25)

 
  at org.kiwix.kiwixmobile.ActivityExtensionsKt.access$getKiwixComponent$p (ActivityExtensionsKt.java:1)

 
  at org.kiwix.kiwixmobile.splash.KiwixSplashActivity.injection (KiwixSplashActivity.java:40)

 
  at org.kiwix.kiwixmobile.core.base.BaseActivity.onCreate (BaseActivity.java:40)

 
  at org.kiwix.kiwixmobile.core.splash.CoreSplashActivity.onCreate (CoreSplashActivity.java:30)

 
  at android.app.Activity.performCreate (Activity.java:7327)

 
  at android.app.Activity.performCreate (Activity.java:7318)

 
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1275)

 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3101)

 
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3264)

 
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:78)

 
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:108)

 
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:68)

 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1955)

 
  at android.os.Handler.dispatchMessage (Handler.java:106)

 
  at android.os.Looper.loop (Looper.java:214)

 
  at android.app.ActivityThread.main (ActivityThread.java:7078)

 
  at java.lang.reflect.Method.invoke (Method.java)

 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:494)

 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:964)
```

--- a/app/src/main/java/org/kiwix/kiwixmobile/ActivityExtensions.kt
+++ b/app/src/main/java/org/kiwix/kiwixmobile/ActivityExtensions.kt
@@ -22,7 +22,16 @@ import org.kiwix.kiwixmobile.core.base.BaseActivity
 import org.kiwix.kiwixmobile.di.components.KiwixComponent
 
 private val BaseActivity.kiwixComponent: KiwixComponent
-  get() = (applicationContext as KiwixApp).kiwixComponent
+  get() = kiwixApp()?.kiwixComponent
+    ?: throw RuntimeException(
+      """
+        applicationContext is ${applicationContext::class.java.simpleName}
+        application is ${application::class.java.simpleName} 
+      """".trimIndent()
+    )
+
+private fun BaseActivity.kiwixApp() =
+  (applicationContext as? KiwixApp) ?: (application as? KiwixApp)
 
 internal inline val BaseActivity.kiwixActivityComponent
   get() = kiwixComponent

--- a/app/src/main/java/org/kiwix/kiwixmobile/ActivityExtensions.kt
+++ b/app/src/main/java/org/kiwix/kiwixmobile/ActivityExtensions.kt
@@ -22,16 +22,14 @@ import org.kiwix.kiwixmobile.core.base.BaseActivity
 import org.kiwix.kiwixmobile.di.components.KiwixComponent
 
 private val BaseActivity.kiwixComponent: KiwixComponent
-  get() = kiwixApp()?.kiwixComponent
-    ?: throw RuntimeException(
-      """
+  get() = kiwixApp()?.kiwixComponent ?: throw RuntimeException(
+    """
         applicationContext is ${applicationContext::class.java.simpleName}
         application is ${application::class.java.simpleName} 
-      """".trimIndent()
-    )
+    """.trimIndent()
+  )
 
-private fun BaseActivity.kiwixApp() =
-  (applicationContext as? KiwixApp) ?: (application as? KiwixApp)
+private fun BaseActivity.kiwixApp() = applicationContext as? KiwixApp ?: application as? KiwixApp
 
 internal inline val BaseActivity.kiwixActivityComponent
   get() = kiwixComponent

--- a/custom/src/main/java/org/kiwix/kiwixmobile/custom/ActivityExtensions.kt
+++ b/custom/src/main/java/org/kiwix/kiwixmobile/custom/ActivityExtensions.kt
@@ -26,11 +26,10 @@ private val BaseActivity.customComponent: CustomComponent
     """
         applicationContext is ${applicationContext::class.java.simpleName}
         application is ${application::class.java.simpleName} 
-    """".trimIndent()
+    """.trimIndent()
   )
 
-private fun BaseActivity.customApp() =
-  (applicationContext as? CustomApp) ?: (application as? CustomApp)
+private fun BaseActivity.customApp() = applicationContext as? CustomApp ?: application as? CustomApp
 
 internal inline val BaseActivity.customActivityComponent
   get() = customComponent.activityComponentBuilder().activity(this).build()

--- a/custom/src/main/java/org/kiwix/kiwixmobile/custom/ActivityExtensions.kt
+++ b/custom/src/main/java/org/kiwix/kiwixmobile/custom/ActivityExtensions.kt
@@ -22,7 +22,15 @@ import org.kiwix.kiwixmobile.core.base.BaseActivity
 import org.kiwix.kiwixmobile.custom.di.CustomComponent
 
 private val BaseActivity.customComponent: CustomComponent
-  get() = (applicationContext as CustomApp).customComponent
+  get() = customApp()?.customComponent ?: throw RuntimeException(
+    """
+        applicationContext is ${applicationContext::class.java.simpleName}
+        application is ${application::class.java.simpleName} 
+    """".trimIndent()
+  )
+
+private fun BaseActivity.customApp() =
+  (applicationContext as? CustomApp) ?: (application as? CustomApp)
 
 internal inline val BaseActivity.customActivityComponent
   get() = customComponent.activityComponentBuilder().activity(this).build()

